interfaces/cython/cantera/onedim.pyx

```diff
@@ -837,6 +928,121 @@ cdef class Sim1D:
         dom, comp = self._get_indices(domain, component)
         self.sim.setFlatProfile(dom, comp, value)
 
+    def collect_data(self, domain, extra):
+        """
+        Return data vector of domain *domain* as `SolutionArray` object
+
+        Derived classes set default values for *domain* and *extra*, where
+        defaults describe flow domain and essential non-thermodynamic solution
+        components of the configuration, respectively. An alternative *domain*
+        (e.g. inlet, outlet, etc.), can be specified either by name or the
+        corresponding Domain1D object itself.
+        """
+        idom = self.domain_index(domain)
+        dom = self.domains[idom]
+
+        extra_cols = OrderedDict()
+        meta = dom.settings
+
+        if isinstance(dom, _FlowBase):
+            n_points = dom.n_points
+
+            # retrieve gas state
+            T = self.profile(self.flame, 'T')
+            P = dom.P
+            k0 = self.flame.component_index(self.gas.species_name(0))
+            Y = [self.profile(self.flame, k)
+                 for k in range(k0, k0 + self.gas.n_species)]
```

Comment:
Can't this be replaced with
```suggestion
            T = self.T
            P = self.P
            Y = self.Y
```
Or just write `states = self.T, self.P, self.Y.T` below?

Reply:
Yes - this is probably a left-over from the previous HDF version where I had to build things from scratch. Only thing is that `self.Y` needs to be transposed.